tracer/src/Datadog.Trace.BenchmarkDotNet/DatadogDiagnoser.cs

```diff
@@ -70,8 +110,235 @@ public void Handle(HostSignal signal, DiagnoserActionParameters parameters)
     /// <inheritdoc />
     public void DisplayResults(ILogger logger)
     {
+        if (_enableProfiler)
+        {
+            if (_platformNotSupportedException is null)
+            {
+                logger.WriteLine($"Datadog Profiler was attached.");
+            }
+            else
+            {
+                logger.WriteLine("Datadog Profiler: " + _platformNotSupportedException.Message);
+            }
+        }
     }
 
     /// <inheritdoc />
     public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters) => Array.Empty<ValidationError>();
+
+    private static IEnumerable<string> GetHomeFolderPaths(DiagnoserActionParameters parameters)
+    {
+        // try to locate it from the environment variable
+        yield return EnvironmentHelpers.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME");
+
+        // try to locate it inside the running benchmark folder
+        yield return Path.Combine(parameters.Process.StartInfo.WorkingDirectory, "datadog");
+
+        // try to locate it inside the benchmark folder
+        yield return Path.Combine(Path.GetDirectoryName(parameters.BenchmarkCase.Descriptor.Type.Assembly.Location) ?? string.Empty, "datadog");
+
+        // try to locate it in the default path using relative path from the benchmark assembly.
+        yield return Path.Combine(
+            Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) ?? string.Empty,
+            "..",
+            "..",
+            "..",
+            "..",
+            "..",
+            "..",
+            "..",
+            "shared",
+            "bin",
+            "monitoring-home");
+    }
+
+    private static bool GetPaths(string monitoringHome, ref string? profiler32Path, ref string? profiler64Path, ref string? loaderConfig, ref string? ldPreload)
```

Comment:
Yes I've seen similar code for the Tracer configuration, in this case is not the same because I only want the profiler supported platforms, meaning no OSX and probably no ARM64